Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift

```diff
@@ -30,7 +30,10 @@ import BraintreeCore
         apiClient.sendAnalyticsEvent(BTAmericanExpressAnalytics.started)
 
         apiClient.get("v1/payment_methods/amex_rewards_balance", parameters: parameters) { [weak self] body, response, error in
-            guard let self = self else { return }
+            guard let self else {
+                self?.notifyFailure(with: BTAmericanExpressError.deallocated, completion: completion)
```

Comment:
I believe this will no-op since `self` is always `nil` in the else block.

Reply:
Yeah, you are right - just tested this. I'll update where we use an optional self to pass the completion and analytics event in the else instead vs using notify. Good catch.

Reply:
+1

Reply:
Updated these here: https://github.com/braintree/braintree_ios/pull/1049/commits/1fadabdbc626829f4dbb6e6decf9fb59066acf2a